### PR TITLE
🐛 Fix checksumming of policies with variants

### PR DESCRIPTION
**Pull request by jaym**
The execution checksum was only taking into account the code id of the query that specified it had variants. This wasn't changing if one of the variants since that query itself has no code.

### CODE REVIEW

File: policy/policy.go [hunk 1]
```diff
@@ -681,3 +689,21 @@ func (s *GroupType) UnmarshalJSON(data []byte) error {
 
 	return errors.New("failed to unmarshal group type: " + str)
 }
+
+func variantsExecutionChecksum(q *explorer.Mquery, c checksums.Fast, includeImpact bool, getQuery func(ctx context.Context, mrn string) (*explorer.Mquery, error)) (checksums.Fast, error) {
+	// TODO: this function needs to be safe against cycles
```
Comment (imilchev) [inline policy/policy.go]: Do you mean that the same `CodeId` shouldn't be added to the checksum multiple times? If yes, then I guess this change should be sufficient:
```go
func variantsExecutionChecksum(q *explorer.Mquery, c checksums.Fast, includeImpact bool, getQuery func(ctx context.Context, mrn string) (*explorer.Mquery, error), added map[string]struct{}) (checksums.Fast, error) {
	if _, ok := added[q.CodeId]; ok {
		return c, nil
	}

	c = c.Add(q.CodeId)
	visited[q.CodeId] = struct{}{}
	if includeImpact {
		c = c.AddUint(q.Impact.Checksum())
	}

	for _, ref := range q.Variants {
		if v, err := getQuery(context.Background(), ref.Mrn); err == nil {
			return variantsExecutionChecksum(v, c, includeImpact, getQuery, added)
		} else {
			return 0, errors.New("cannot find dependent composed query '" + ref.Mrn + "'")
		}
	}
	return c, nil
}
```

Comment (jaym) [reply policy/policy.go]: its not really about being added so much as we dont want cycles because then the function wouldn't terminate

Comment (jaym) [reply policy/policy.go]: also, its possible that there is not a code id because there is no code

File: policy/policy.go [hunk 2]
```diff
@@ -681,3 +689,21 @@ func (s *GroupType) UnmarshalJSON(data []byte) error {
...
+	c = c.
+		Add(q.CodeId)
+	if includeImpact {
+		c = c.AddUint(q.Impact.Checksum())
+	}
+
+	for _, ref := range q.Variants {
+		if v, err := getQuery(context.Background(), ref.Mrn); err == nil {
+			return variantsExecutionChecksum(v, c, includeImpact, getQuery)
```
Comment (jaym) [inline policy/policy.go]: this should not return

File: policy/policy.go (hunk 1 above)
Comment (jaym) [reply policy/policy.go]: There are other things before this code runs which assume there are no cycles. So this problem probably needs to be fixed far before we get here

**Review by chris-rock — APPROVED** [review]
Thank you @jaym Looks good to me

**Pull request merged**